Source/JavaScriptCore/wasm/WasmBBQJIT.cpp

```diff
@@ -5920,52 +5926,106 @@ class BBQJIT {
         });
         m_jit.move(m_scratchGPR, MacroAssembler::stackPointerRegister);
 
+        LocalOrTempIndex i = 0;
+        for (; i < m_arguments.size(); ++i)
+            flushValue(Value::fromLocal(m_parser->typeOfLocal(i).kind, i));
+
         // Zero all locals that aren't initialized by arguments.
-        // This is kind of icky...we can evaluate replacing this a memset() or a tracker for which
-        // locals have been initialized.
-        for (LocalOrTempIndex i = 0; i < m_locals.size(); ++i) {
-            if (i < m_arguments.size()) {
-                flushValue(Value::fromLocal(m_parser->typeOfLocal(i).kind, i));
-                continue;
+        enum class ClearMode { Zero, JSNull };
+        std::optional<int32_t> highest;
+        std::optional<int32_t> lowest;
+        auto flushZeroClear = [&]() {
+            if (!lowest)
+                return;
+            size_t size = highest.value() - lowest.value();
+            int32_t pointer = lowest.value();
+
+            // Adjust pointer offset to be efficient for paired-stores.
+            if (pointer & 4 && size >= 4) {
+                m_jit.store32(TrustedImm32(0), Address(GPRInfo::callFrameRegister, pointer));
+                pointer += 4;
+                size -= 4;
+            }
+            if (pointer & 8 && size >= 8) {
+                m_jit.store64(TrustedImm64(0), Address(GPRInfo::callFrameRegister, pointer));
+                pointer += 8;
+                size -= 8;
             }
 
-            Value zero;
-            switch (m_parser->typeOfLocal(i).kind) {
+#if CPU(ARM64)
+            unsigned count = size / 16;
+            for (unsigned i = 0; i < count; ++i) {
+                m_jit.storePair64(ARM64Registers::zr, ARM64Registers::zr, GPRInfo::callFrameRegister, TrustedImm32(pointer));
+                pointer += 16;
+                size -= 16;
+            }
+#else
+            unsigned count = size / 8;
+            for (unsigned i = 0; i < count; ++i) {
+                m_jit.store64(TrustedImm64(0), Address(GPRInfo::callFrameRegister, pointer));
+                pointer += 8;
+                size -= 8;
+            }
+#endif
+            // Fill remaining part.
+            if (size & 8) {
+                m_jit.store64(TrustedImm64(0), Address(GPRInfo::callFrameRegister, pointer));
+                pointer += 8;
+                size -= 8;
+            }
```

Comment:
I think you can move this under the `CPU(ARM64)` section because `(size & 8)` should never be true after the `unsigned count = size / 8;` loop.

Reply:
Changed.